### ISSUE TITLE
[Mercenaries For Me] Unity Engine crashes, semi random. Related to Mercenaries for Me

### ISSUE THREAD

**Issue by Sinuisan** [cseelhoff/RimThreaded#471]
**Describe the bug**
Game crashes and creates a unity crash log. Save is can be resumed on reboot. 

**Steps to reproduce the behavior (VERY IMPORTANT)**
Seems pretty random, I think its related to an event or world action Mercenary for Me is trying to activate, might be reproduced using dev commands. 

**Error Log**
https://pastebin.com/ChKsiDV5  - PlayerLog
https://gist.github.com/7e5df234c5e5be630e84941dcee09a63 - HugsLib from reloaded Save

**Save file**
[Autosave-4.zip](https://github.com/cseelhoff/RimThreaded/files/6043021/Autosave-4.zip)

**Screenshots**
N/A

**Mod list (Preferably a RimPy compatible list.)**

Taken From Fluffy as an XML. Hope thats useful? Ordered according to the RimThreaded wiki guide.
 
<?xml version="1.0" encoding="UTF-8"?>

-<ModList>

<Name>Post re-ordering</Name>

-<modIds>

<li>brrainz.harmony</li>

<li>fluffy.modmanager</li>

<li>me.samboycoding.betterloading</li>

<li>ludeon.rimworld</li>

<li>vanillaexpanded.achievements</li>

<li>unlimitedhugs.hugslib</li>

<li>jecrell.jecstools</li>

<li>erdelf.humanoidalienraces</li>

<li>edb.preparecarefully</li>

<li>oskarpotocki.vanillafactionsexpanded.core</li>

<li>roolo.giddyupcore</li>

<li>roolo.giddyupbattlemounts</li>

<li>roolo.giddyupcaravan</li>

<li>unlimitedhugs.mapreroll</li>

<li>cabbage.rimcities</li>

<li>woolstrand.realruins</li>

<li>arandomkiwi.mercenariesforme</li>

<li>oskarpotocki.vfe.insectoid</li>

<li>oskarpotocki.vfe.mechanoid</li>

<li>vanillaexpanded.vwe</li>

<li>oskarpotocki.vanillafactionsexpanded.medievalmodule</li>

<li>oskarpotocki.vanillafactionsexpanded.settlersmodule</li>

<li>vanillaexpanded.vanillatraitsexpanded</li>

<li>atlas.androidtiers</li>

<li>arandomkiwi.androidtierstxserie</li>

<li>malistaticy.androidresurrection</li>

<li>troopersmith1.deathrattle</li>

<li>troopersmith1.lifesupport</li>

<li>vat.epoeforked</li>

<li>kikohi.cybernetic</li>

<li>community.psychology.unofficialupdate</li>

<li>orion.hospitality</li>

<li>dubwise.dubsbadhygiene</li>

<li>petetimessix.simplesidearms</li>

<li>uuugggg.smartmedicine</li>

<li>dhultgren.smarterconstruction</li>

<li>codeoptimist.jobsofopportunity</li>

<li>mlie.billysimprovedcaravanformation</li>

<li>avilmask.commonsense</li>

<li>dubwise.dubsbreakmod</li>

<li>weilbyte.snapout</li>

<li>roolo.runandgun</li>

<li>roolo.searchanddestroy</li>

<li>captainmuscles.nopawnleftbehind</li>

<li>brrainz.achtung</li>

<li>crlsniper.realisticrooms</li>

<li>brrainz.cameraplus</li>

<li>longercfloor.proxyheat</li>

<li>jaxe.foodrestrictions</li>

<li>syrchalis.setupcamp</li>

<li>jelly.directabandonsettlement</li>

<li>hatti.qualitybuilder</li>

<li>vanillaexpanded.vee</li>

<li>fluffy.fluffybreakdowns</li>

<li>vinalx.researchpalforked</li>

<li>fluffy.medicaltab</li>

<li>fluffy.pharmacist</li>

<li>jaxe.rimhud</li>

<li>fluffy.blueprints</li>

<li>com.github.alandariva.moreplanning</li>

<li>unlimitedhugs.allowtool</li>

<li>jaxe.toggleharvest</li>

<li>automatic.bionicicons</li>

<li>owlchemist.bettervanillamasking</li>

<li>jaxe.bubbles</li>

<li>bodlosh.weaponstats</li>

<li>sandy.rpgstyleinventory</li>

<li>saucypigeon.pawnbadge</li>

<li>dubwise.dubsmintmenus</li>

<li>petetimessix.compacthediffs</li>

<li>timmyliang.tradehelper</li>

<li>longwater.smartstockpilecreation</li>

<li>dubwise.dubsmintminimap</li>

<li>mlie.storagesearch</li>

<li>captainmuscles.categorizedbills</li>

<li>derekbickley.ltocolonygroupsfinal</li>

<li>ghxx.techadvancing</li>

<li>sineswiper.xenobionicpatcher</li>

<li>mlie.lessarbitrarysurgery</li>

<li>hobtook.mortaraccuracy</li>

<li>improvedloadtransportpods.rw</li>

<li>mlie.variousspaceshipchunk</li>

<li>dubwise.rimatomics</li>

<li>dubwise.rimefeller</li>

<li>zylle.gemstones</li>

<li>kikohi.jewelry</li>

<li>wemd.fueledheaters</li>

<li>fluffy.colonymanager</li>

<li>fluffy.backuppower</li>

<li>antaioz.bettervents</li>

<li>dubwise.dubspaintshop</li>

<li>kentington.saveourship2</li>

<li>smashphil.neceros.srtsexpanded</li>

<li>neceros.srtserealismpatch</li>

<li>sarg.alphaships</li>

<li>torann.rimwar</li>

<li>ratys.rtfuse</li>

<li>tradingspot.kv.rw</li>

<li>nephlite.orbitaltradecolumn</li>

<li>rimfridge.kv.rw</li>

<li>soulfulpumpkin.tabletoptrove</li>

<li>fluffy.stuffedfloors</li>

<li>mlie.simplymorebridges</li>

<li>lwm.deepstorage</li>

<li>mlie.pickupandhaul</li>

<li>dracoix.doormat.r12a</li>

<li>wemd.reinforceddoors</li>

<li>neceros.reinforcedwalls</li>

<li>tidal.reinforcedstructurespatch</li>

<li>unlimitedhugs.remotetech</li>

<li>vanillaexpanded.vfecore</li>

<li>vanillaexpanded.vfeart</li>

<li>vanillaexpanded.vfefarming</li>

<li>vanillaexpanded.vfemedical</li>

<li>vanillaexpanded.vfepower</li>

<li>vanillaexpanded.vfeproduction</li>

<li>vanillaexpanded.vfesecurity</li>

<li>n7huntsman.palisades</li>

<li>neceros.mud</li>

<li>vanillaexpanded.vfespacer</li>

<li>dubwise.dubsrimkit</li>

<li>halo.rsf</li>

<li>vanillaexpanded.vappe</li>

<li>vanillaexpanded.vhe</li>

<li>vanillaexpanded.vhe.beards</li>

<li>vanillaexpanded.varme</li>

<li>mlie.cyberneticwarfare</li>

<li>vanillaexpanded.vweq</li>

<li>vanillaexpanded.vwel</li>

<li>vanillaexpanded.vweg</li>

<li>vanillaexpanded.vwec</li>

<li>kikohi.advancedparka</li>

<li>vanillaexpanded.vcef</li>

<li>vanillaexpanded.vbookse</li>

<li>vanillaexpanded.vplantse</li>

<li>vanillaexpanded.vplantsesucculents</li>

<li>vanillaexpanded.vcookesushi</li>

<li>vanillaexpanded.vcookestews</li>

<li>vanillaexpanded.vcooke</li>

<li>vanillaexpanded.vbrewecandt</li>

<li>vanillaexpanded.vbrewe</li>

<li>owlchemist.undergroundpowerconduits</li>

<li>murmur.walllight</li>

<li>sumghai.medpod</li>

<li>notfood.mendandrecycle</li>

<li>heremeus.medicaldissection</li>

<li>pyrce.mass.graves</li>

<li>averythekitty.tillableterrains</li>

<li>vanillaexpanded.vaeas</li>

<li>vanillaexpanded.vaeau</li>

<li>vanillaexpanded.vaebf</li>

<li>vanillaexpanded.vaecd</li>

<li>vanillaexpanded.vaed</li>

<li>vanillaexpanded.vaeendandext</li>

<li>vanillaexpanded.vaeed</li>

<li>vanillaexpanded.vaeis</li>

<li>vanillaexpanded.vaetf</li>

<li>vanillaexpanded.vaetr</li>

<li>vanillaexpanded.vaets</li>

<li>vanillaexpanded.vaet</li>

<li>vanillaexpanded.vtexe</li>

<li>ogre.ogrestack</li>

<li>rikiki.miningco.mms</li>

<li>syrchalis.bulletcasings</li>

<li>sneaks.meatblend</li>

<li>jaxe.morerandomseeds</li>

<li>majorhoff.rimthreaded</li>

</modIds>

-<modNames>

<li>Harmony</li>

<li>Mod Manager</li>

<li>BetterLoading</li>

<li>Core</li>

<li>Vanilla Achievements Expanded</li>

<li>HugsLib</li>

<li>JecsTools (Unofficial)</li>

<li>Humanoid Alien Races 2.0</li>

<li>EdB Prepare Carefully</li>

<li>Vanilla Expanded Framework</li>

<li>Giddy-up! Core</li>

<li>Giddy-up! Battle Mounts</li>

<li>Giddy-up! Caravan</li>

<li>Map Reroll</li>

<li>RimCities - Citadel Update</li>

<li>Real Ruins</li>

<li>Mercenaries For Me</li>

<li>Vanilla Factions Expanded - Insectoids</li>

<li>Vanilla Factions Expanded - Mechanoids</li>

<li>Vanilla Weapons Expanded</li>

<li>Vanilla Factions Expanded - Medieval</li>

<li>Vanilla Factions Expanded - Settlers</li>

<li>Vanilla Traits Expanded</li>

<li>Android tiers</li>

<li>Android tiers - TX Series</li>

<li>Android Resurrection Kit (rebuild)</li>

<li>Death Rattle Continued [1.2]</li>

<li>Life Support Continued [1.1][1.2]</li>

<li>Expanded Prosthetics and Organ Engineering - Forked</li>

<li>Cybernetic Organism and Neural Network</li>

<li>Psychology (unofficial 1.1/1.2) </li>

<li>Hospitality</li>

<li>Dubs Bad Hygiene</li>

<li>Simple sidearms</li>

<li>Smart Medicine</li>

<li>Smarter Construction</li>

<li>Jobs of Opportunity (While You're Up)</li>

<li>Billy's Improved Caravan Formation (Continued)</li>

<li>Common Sense</li>

<li>Dubs Break Mod</li>

<li>Snap Out!</li>

<li>RunAndGun</li>

<li>Search and Destroy</li>

<li>No One Left Behind</li>

<li>Achtung!</li>

<li>Realistic Rooms</li>

<li>Camera+</li>

<li>Proxy Heat</li>

<li>Animal Food Restrictions</li>

<li>[SYR] Set Up Camp</li>

<li>Directly Abandon Settlement</li>

<li>QualityBuilder</li>

<li>Vanilla Events Expanded</li>

<li>Fluffy Breakdowns</li>

<li>ResearchPal - Forked</li>

<li>Medical Tab</li>

<li>Pharmacist</li>

<li>RimHUD</li>

<li>Blueprints</li>

<li>More Planning</li>

<li>Allow Tool</li>

<li>Toggle Harvest</li>

<li>Bionic icons</li>

<li>Better Vanilla Masking</li>

<li>Interaction Bubbles</li>

<li>WeaponStats</li>

<li>RPG Style Inventory</li>

<li>Pawn Badge Fan Fork</li>

<li>Dubs Mint Menus</li>

<li>Compact Hediffs</li>

<li>TradeHelper</li>

<li>Quick Stockpile Creation</li>

<li>Dubs Mint Minimap</li>

<li>Rimworld Search Agency (Continued)</li>

<li>Categorized Crafting Bills</li>

<li>[LTO] Colony Groups</li>

<li>Tech Advancing</li>

<li>Xenobionic Patcher</li>

<li>Less Arbitrary Surgery (Continued)</li>

<li>Mortar Accuracy</li>

<li>Improved Load Transport Pods</li>

<li>Various Space Ship Chunk (Continued)</li>

<li>Rimatomics</li>

<li>Rimefeller</li>

<li>Gemstones</li>

<li>Jewelry</li>

<li>[WD] Fueled Heaters</li>

<li>Colony Manager</li>

<li>Backup Power</li>

<li>BetterVents</li>

<li>Dub's Paint Shop</li>

<li>Save Our Ship 2</li>

<li>SRTS Expanded</li>

<li>SRTS Expanded Realism Patch</li>

<li>Alpha Ships</li>

<li>Rim War</li>

<li>RT Fuse</li>

<li>[KV] Trading Spot</li>

<li>Utility Columns</li>

<li>[KV] RimFridge</li>

<li>Tabletop Trove - Additional Joy Objects and Decor</li>

<li>Stuffed Floors</li>

<li>Simply More Bridges (Continued)</li>

<li>LWM's Deep Storage</li>

<li>Pick Up And Haul (Continued)</li>

<li>Door Mat</li>

<li>[WD] Reinforced Doors</li>

<li>Reinforced Walls</li>

<li>[WD] Reinforced Doors + Reinforced Walls Patch</li>

<li>Remote Tech</li>

<li>Vanilla Furniture Expanded</li>

<li>Vanilla Furniture Expanded - Art</li>

<li>Vanilla Furniture Expanded - Farming</li>

<li>Vanilla Furniture Expanded - Medical Module</li>

<li>Vanilla Furniture Expanded - Power</li>

<li>Vanilla Furniture Expanded - Production</li>

<li>Vanilla Furniture Expanded - Security</li>

<li>Palisades</li>

<li>Mud</li>

<li>Vanilla Furniture Expanded - Spacer Module</li>

<li>Dubs Rimkit</li>

<li>Rimworld: Spartan Foundry</li>

<li>Vanilla Apparel Expanded</li>

<li>Vanilla Hair Expanded</li>

<li>Vanilla Hair Expanded - Beards</li>

<li>Vanilla Armour Expanded</li>

<li>Cybernetic Warfare and Special Weapons (Continued)</li>

<li>Vanilla Weapons Expanded - Quickdraw</li>

<li>Vanilla Weapons Expanded - Laser</li>

<li>Vanilla Weapons Expanded - Grenades</li>

<li>Vanilla Weapons Expanded - Coilguns</li>

<li>Advanced Parka</li>

<li>Vanilla Fishing Expanded</li>

<li>Vanilla Books Expanded</li>

<li>Vanilla Plants Expanded</li>

<li>Vanilla Plants Expanded - Succulents</li>

<li>Vanilla Cooking Expanded - Sushi</li>

<li>Vanilla Cooking Expanded - Stews</li>

<li>Vanilla Cooking Expanded</li>

<li>Vanilla Brewing Expanded - Coffees and Teas</li>

<li>Vanilla Brewing Expanded</li>

<li>Underground Power Conduits</li>

<li>Wall Light</li>

<li>MedPod</li>

<li>MendAndRecycle</li>

<li>Medical Dissection</li>

<li>Mass Graves</li>

<li>Tillable Terrains [VPE Patch]</li>

<li>Vanilla Animals Expanded — Arid Shrubland</li>

<li>Vanilla Animals Expanded — Australia</li>

<li>Vanilla Animals Expanded — Boreal Forest</li>

<li>Vanilla Animals Expanded — Cats and Dogs</li>

<li>Vanilla Animals Expanded — Desert</li>

<li>Vanilla Animals Expanded — Endangered</li>

<li>Vanilla Animals Expanded — Extreme Desert</li>

<li>Vanilla Animals Expanded — Ice Sheet</li>

<li>Vanilla Animals Expanded — Temperate Forest</li>

<li>Vanilla Animals Expanded — Tropical Rainforest</li>

<li>Vanilla Animals Expanded — Tropical Swamp</li>

<li>Vanilla Animals Expanded — Tundra</li>

<li>Vanilla Textures Expanded</li>

<li>OgreStack</li>

<li>MiningCo. MMS</li>

<li>[SYR] Bullet Casings</li>

<li>Meat Blend</li>

<li>More Random Seeds</li>

<li>RimThreaded [1.2]</li>

</modNames>

</ModList>

**Comment by Sinuisan** [comment]
Small update, disabling events via the mod options for Merc for Me seems so far to have stopped the bug from appearing. Suggest its a temporary fix for others.

**Comment by Eldulac** [comment]
**[Tested on RT 1.3.2]**

I tested this issue on the save file posted here, as well as two fresh games. The attached save file was extremely unstable and threw constant red errors. The load order and mod list were problematic and contained multiple incompatible mods.

Despite repeated attempts, I was not able to find any problems with "Mercenaries for Me"- mod, even though I triggered the incidents of the mod through dev- mode, and passed long periods of time in-game. There was one incident though, the "Counter Offer", that I was not able to trigger during the test. This may have been the cause of the issue, as triggering it seems to be based on chance.

It is worth noting that the "Mercenaries for me"- mod had been updated since this issue was encountered, which may have fixed the issue.

**Conclusion**
The test was inconclusive, as I was unable to trigger the "Counter Offer" incident. No errors were encountered regarding "Mercenaries for me" mod, and all the features of the mod seemed to work on a fresh game.
